### PR TITLE
Copter: disable first set of useless features

### PR DESCRIPTION
**Pull request by khancyr**
I have removed some features I am pretty sure we don't use.

### CODE REVIEW

File: ArduCopter/APM_Config.h [hunk 1]
```diff
@@ -5,44 +5,44 @@
...
-//#define MODE_DRIFT_ENABLED    DISABLED            // disable drift mode support
-//#define MODE_FLIP_ENABLED     DISABLED            // disable flip mode support
-//#define MODE_FOLLOW_ENABLED   DISABLED            // disable follow mode support
+#define MODE_CIRCLE_ENABLED   DISABLED            // disable circle mode support
+#define MODE_DRIFT_ENABLED    DISABLED            // disable drift mode support
+#define MODE_FLIP_ENABLED     DISABLED            // disable flip mode support
+#define MODE_FOLLOW_ENABLED   DISABLED            // disable follow mode support
 //#define MODE_GUIDED_ENABLED   DISABLED            // disable guided mode support
-//#define MODE_GUIDED_NOGPS_ENABLED   DISABLED      // disable guided/nogps mode support
+#define MODE_GUIDED_NOGPS_ENABLED   DISABLED      // disable guided/nogps mode support
```
Comment (Charlie-Burge) [inline ArduCopter/APM_Config.h:36]: We'll need to re-enable GUIDED no GPS

**Review by Charlie-Burge — CHANGES_REQUESTED** [review]
LGTM - just the one change for a mode we do use

File: ArduCopter/APM_Config.h (hunk 1 above)
Comment (khancyr) [reply ArduCopter/APM_Config.h:36]: That is unexpected ! I will revert.
Anything else ?

File: ArduCopter/APM_Config.h [hunk 2]
```diff
@@ -5,44 +5,44 @@
...
 //#define RANGEFINDER_ENABLED   DISABLED            // disable rangefinder to save 1k of flash
 //#define AC_RALLY              DISABLED            // disable rally points library (must also disable terrain which relies on rally)
 //#define AC_AVOID_ENABLED      DISABLED            // disable stop-at-fence library
 //#define AC_OAPATHPLANNER_ENABLED DISABLED         // disable path planning around obstacles
 //#define AC_TERRAIN            DISABLED            // disable terrain library
 //#define PARACHUTE             DISABLED            // disable parachute release to save 1k of flash
 //#define NAV_GUIDED            DISABLED            // disable external navigation computer ability to control vehicle through MAV_CMD_NAV_GUIDED mission commands
-//#define OPTFLOW               DISABLED            // disable optical flow sensor to save 5K of flash space
+#define OPTFLOW               DISABLED            // disable optical flow sensor to save 5K of flash space
 //#define FRSKY_TELEM_ENABLED   DISABLED            // disable FRSky telemetry
```
Comment (Dan-Moss) [inline ArduCopter/APM_Config.h:20]: dont beleive we use this?

File: ArduCopter/APM_Config.h [hunk 3]
```diff
@@ -5,44 +5,44 @@
...
-//#define PRECISION_LANDING     DISABLED            // disable precision landing using companion computer or IRLock sensor
-//#define BEACON_ENABLED        DISABLED            // disable beacon support
-//#define SPRAYER_ENABLED       DISABLED            // disable the crop sprayer feature (two ESC controlled pumps the speed of which depends upon the vehicle's horizontal velocity)
+#define PRECISION_LANDING     DISABLED            // disable precision landing using companion computer or IRLock sensor
+#define BEACON_ENABLED        DISABLED            // disable beacon support
+#define SPRAYER_ENABLED       DISABLED            // disable the crop sprayer feature (two ESC controlled pumps the speed of which depends upon the vehicle's horizontal velocity)
 //#define WINCH_ENABLED         DISABLED            // disable winch support
```
Comment (Dan-Moss) [inline ArduCopter/APM_Config.h:24]: not using this yet

File: ArduCopter/APM_Config.h [hunk 4]
```diff
@@ -5,44 +5,44 @@
...
-//#define GRIPPER_ENABLED       DISABLED            // disable gripper support
+#define GRIPPER_ENABLED       DISABLED            // disable gripper support
 //#define RPM_ENABLED           DISABLED            // disable rotations per minute sensor support
 //#define STATS_ENABLED         DISABLED            // disable statistics support
 //#define MODE_ACRO_ENABLED     DISABLED            // disable acrobatic mode support
```
Comment (Dan-Moss) [inline ArduCopter/APM_Config.h:28]: no use for this?

File: ArduCopter/APM_Config.h [hunk 5]
```diff
@@ -5,44 +5,44 @@
...
+#define MODE_DRIFT_ENABLED    DISABLED            // disable drift mode support
+#define MODE_FLIP_ENABLED     DISABLED            // disable flip mode support
+#define MODE_FOLLOW_ENABLED   DISABLED            // disable follow mode support
 //#define MODE_GUIDED_ENABLED   DISABLED            // disable guided mode support
-//#define MODE_GUIDED_NOGPS_ENABLED   DISABLED      // disable guided/nogps mode support
+#define MODE_GUIDED_NOGPS_ENABLED   DISABLED      // disable guided/nogps mode support
 //#define MODE_LOITER_ENABLED   DISABLED            // disable loiter mode support
 //#define MODE_POSHOLD_ENABLED  DISABLED            // disable poshold mode support
 //#define MODE_RTL_ENABLED      DISABLED            // disable rtl mode support
 //#define MODE_SMARTRTL_ENABLED DISABLED            // disable smartrtl mode support
```
Comment (Dan-Moss) [inline ArduCopter/APM_Config.h:40]: we no longer use SRTL

**Review by Dan-Moss — COMMENTED** [review]
added some comments on modes we dont use

File: ArduCopter/APM_Config.h (hunk 5 above)
Comment (khancyr) [reply ArduCopter/APM_Config.h:40]: that is great thing, it saves a lot

File: ArduCopter/APM_Config.h [hunk 6]
```diff
@@ -5,44 +5,44 @@
...
+#define MODE_SPORT_ENABLED    DISABLED            // disable sport mode support
 //#define MODE_SYSTEMID_ENABLED DISABLED            // disable system ID mode support
-//#define MODE_THROW_ENABLED    DISABLED            // disable throw mode support
-//#define MODE_ZIGZAG_ENABLED   DISABLED            // disable zigzag mode support
-//#define OSD_ENABLED           DISABLED            // disable on-screen-display support
+#define MODE_THROW_ENABLED    DISABLED            // disable throw mode support
+#define MODE_ZIGZAG_ENABLED   DISABLED            // disable zigzag mode support
+#define OSD_ENABLED           DISABLED            // disable on-screen-display support
 //#define BUTTON_ENABLED        DISABLED            // disable button support
 //#define LANDING_GEAR_ENABLED  DISABLED            // disable landing gear support
```
Comment (Charlie-Burge) [inline ArduCopter/APM_Config.h:47]: We also don't use retractable landing gear so could disable this one

File: ArduCopter/APM_Config.h (hunk 1 above)
Comment (Gone4Dirt) [reply ArduCopter/APM_Config.h:36]: You will need to re-enable flip as well ;-)

Comment (khancyr) [reply ArduCopter/APM_Config.h:36]: that is courageous to do flip with manna drone considering the flip code !

**Pull request closed without merging**